src/eos-shard-writer.c

```diff
@@ -99,6 +99,23 @@ eos_shard_writer_init (EosShardWriter *self)
   g_array_set_clear_func (self->entries, (GDestroyNotify) eos_shard_writer_record_entry_clear);
 }
 
+void
+insert_record (EosShardWriter *self,
+               char *hex_name,
+               int index)
+{
+  struct eos_shard_writer_record_entry record_entry = {};
+
+  eos_shard_util_hex_name_to_raw_name (record_entry.raw_name, hex_name);
+
+  if (index > 0) {
+    struct eos_shard_writer_record_entry *e = &g_array_index (self->entries, struct eos_shard_writer_record_entry, index - 1);
+    g_assert (memcmp (record_entry.raw_name, e->raw_name, EOS_SHARD_RAW_NAME_SIZE) > 0);
+  }
```

Comment:
We should verify that we're greater than what came before it, and less than what comes after it -- that is, that this is a proper sorted insert.